### ISSUE TITLE
Rollover Items for Operations Report Revamp - Low Priority Items

### ISSUE THREAD

**Issue by robertsamm** [Analyticsphere/metricsReportsRequests#164]
Low priority items for Operations Report Revamp, do not need to be completed prior to generating report in prod and posting to ALL:

1. Review and potential exclude extreme value in Figure 1.10
2. Adjust 'Total' column percentages in Tables: 1.14, 1.17, 4.15-4.18, 6.2
3. Adjust cell percentages in Table 1.6
4. Discuss high HF missings in Table 1.4 with Madhuri and potentially update the category logic
5. Adjust the 'Total' rows in Section 2
6. Add alternating shading to larger tables to make it easier to review data across rows

**Comment by KELSEYDOWLING7** [comment]
- [x] Review and potential exclude extreme value in Figure 1.10
- [x] Adjust 'Total' column to column percentages in Tables: 1.14, 1.17, and 6.2
- [x] Adjust 'Total' column to column percentages in Tables: 4.15-4.18
- [x] Adjust cell percentages in Table 1.6
- [x] Discuss high HF missings in Table 1.4 with Madhuri and potentially update the category logic
- [x] Adjust the 'Total' rows in Section 2
- [x] Add alternating shading to larger tables to make it easier to review data across rows
- [x] Capitalization or Spelling errors

**Comment by KELSEYDOWLING7** [comment]
To confirm the request here, you want the row percentage in the other columns to remain the same, but in the total row have the percentage be a column percentage? Please note the that the row percentages would add up to 100%, so that may get confusing. For example, in the first row we would have 1,840 (84.91%), 178 (8.21%), 149 (6.88%), 2,167 (60.36%)
![image](https://github.com/user-attachments/assets/2a36485c-4a6e-45cb-9fea-0b6865bf4a49)

**Comment by KELSEYDOWLING7** [comment]
To clarify the request here, do you want the Signed-in column to be 100%, and then the remaining percentages be the amount of the previous column? For example, the first two rows would be:
Baylor Scott and White 14 (100%), 13(92.86), 13(92.86), 13(92.86), 13(92.86), 287
HealthPartners 10,230 (100%), 8,489 (82.98%), 8,252 (97.21%), 8,240 (99.85%), 7,931 (96.25%), 109,912
![image](https://github.com/user-attachments/assets/18acf409-f2c6-4a8b-941c-7c75063f33ee)

**Comment by robertsamm** [comment]
@KELSEYDOWLING7 see responses below

> To confirm the request here, you want the row percentage in the other columns to remain the same, but in the total row have the percentage be a column percentage? Please note the that the row percentages would add up to 100%, so that may get confusing. For example, in the first row we would have 1,840 (84.91%), 178 (8.21%), 149 (6.88%), 2,167 (60.36%) ![image](https://private-user-images.githubusercontent.com/101136647/367315953-2a36485c-4a6e-45cb-9fea-0b6865bf4a49.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjYyNDk0MDYsIm5iZiI6MTcyNjI0OTEwNiwicGF0aCI6Ii8xMDExMzY2NDcvMzY3MzE1OTUzLTJhMzY0ODVjLTRhNmUtNDVjYi05ZmVhLTBiNjg2NWJmNGE0OS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwOTEzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDkxM1QxNzM4MjZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1mMDYzMmUwMjc4Njg0ZmViYWRkYjZkMTI0MWViY2EzYWU2YzFjZjFkNDUxODk2ZWM2NzgwNjY2YWNlNThlYmNiJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.guDKg7UsRM24inf8r4mH4EBSyFsUiBFZmxfS3P2GhMs)

For this one - the inner cells should change to cell percents and then the total row would have a row percent and the total column would be a column percent.

> To clarify the request here, do you want the Signed-in column to be 100%, and then the remaining percentages be the amount of the previous column? For example, the first two rows would be: Baylor Scott and White 14 (100%), 13(92.86), 13(92.86), 13(92.86), 13(92.86), 287 HealthPartners 10,230 (100%), 8,489 (82.98%), 8,252 (97.21%), 8,240 (99.85%), 7,931 (96.25%), 109,912 ![image](https://private-user-images.githubusercontent.com/101136647/367320344-18acf409-f2c6-4a8b-941c-7c75063f33ee.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjYyNDk0MDYsIm5iZiI6MTcyNjI0OTEwNiwicGF0aCI6Ii8xMDExMzY2NDcvMzY3MzIwMzQ0LTE4YWNmNDA5LWYyYzYtNGE4Yi05NDFjLTdjNzUwNjNmMzNlZS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwOTEzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDkxM1QxNzM4MjZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0wMzFjOTVkMWVmMzI5YTk2OTYyNTdlMDlkMWIzNmQwNTJmYTNmZDFlZWRjNjFmZGQwOWJmZjE3MmYyNmM2ZTMxJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.PfU0A069HzdqYMveDLZ3bNKyzzv-CpGg16kdOaB5tXI)

For this one, yes, that sounds exactly right!

**Comment by brotzmanmj** [comment]
Please also have someone (perhaps Leila?) check the report throughout carefully for spelling errors. I noticed a few at a glance in the report in prod today, so it just needs a thorough review. Thanks.

**Comment by KELSEYDOWLING7** [comment]
Hi @lorszag would you have time to review the Operations report we discussed last Friday for spelling errors? The report is posted here, and please let me know if you have any issues accessing it: https://nih.app.box.com/file/1648619607764

**Comment by lorszag** [comment]
@KELSEYDOWLING7 absolutely. I should have time later this morning/early afternoon.

**Comment by lorszag** [comment]
@KELSEYDOWLING7 Mostly just capitalization errors. Happy to go through the code myself and make the changes if that would be easier? Let me know.

**Comment by KELSEYDOWLING7** [comment]
@lorszag Thanks so much! With the Github pushing and pulling and multiple branches it might be easier for me to do it locally but I appreciate it. If you have a list do you want to send it to me? Or were you able to comment on the pdf directly?

**Comment by brotzmanmj** [comment]
@KELSEYDOWLING7 and @lorszag let us know if you have a run of the cleaned up report for us to review before it goes live again on Monday? I mean just for the spelling and capitalization corrections, not expecting the other issues to be updated yet. Thanks.

**Comment by KELSEYDOWLING7** [comment]
@brotzmanmj  Yes, I can run that now. The issues checked off above will be applied to this version as well

**Comment by KELSEYDOWLING7** [comment]
There's a new error message popping up. I'm on PTO tomorrow, so if I can't get it to run tonight I will run the report first thing on Monday morning

**Comment by KELSEYDOWLING7** [comment]
@brotzmanmj  Here is the version with the capitalization and spelling errors. Mostly capitalization was found so please let us know if we missed anything.

Also I fixed the plot issue. The only plots I could not fix were Figures 6.13 and 6.14

https://nih.app.box.com/file/1655448666217

**Comment by brotzmanmj** [comment]
@KELSEYDOWLING7  Thanks for restoring this! Figure 6.1 still has a spelling error on y axis. I'll review the rest tomorrow. Thanks

**Comment by KELSEYDOWLING7** [comment]
@mnataraj92  Do you know why these numbers may be off or if we're missing something in the logic? It looks like it's mostly a HF problem. Not sure if the QC reflects this as well

- **Automated Verification**: verified=yes, RcrtV_Automated_v1r0=method used, RcrtV_Mannual_v1r0= method NOT used, RcrtV_Outreach_v1r0 = No/NULL

- **Manual Verification**: verified=yes, RcrtV_Mannual_v1r0= method used, RcrtV_Outreach_v1r0 =NO/Null

- **Manual Verification and Outreach**: verified=yes, RcrtV_Mannual_v1r0= method used, RcrtV_Outreach_v1r0=yes


![image](https://github.com/user-attachments/assets/88782442-a4f8-45da-8492-363e8c0a3426)

**Comment by KELSEYDOWLING7** [comment]
> @KELSEYDOWLING7 see responses below
> 
> > To confirm the request here, you want the row percentage in the other columns to remain the same, but in the total row have the percentage be a column percentage? Please note the that the row percentages would add up to 100%, so that may get confusing. For example, in the first row we would have 1,840 (84.91%), 178 (8.21%), 149 (6.88%), 2,167 (60.36%) ![image](https://private-user-images.githubusercontent.com/101136647/367315953-2a36485c-4a6e-45cb-9fea-0b6865bf4a49.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjYyNDk0MDYsIm5iZiI6MTcyNjI0OTEwNiwicGF0aCI6Ii8xMDExMzY2NDcvMzY3MzE1OTUzLTJhMzY0ODVjLTRhNmUtNDVjYi05ZmVhLTBiNjg2NWJmNGE0OS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwOTEzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDkxM1QxNzM4MjZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1mMDYzMmUwMjc4Njg0ZmViYWRkYjZkMTI0MWViY2EzYWU2YzFjZjFkNDUxODk2ZWM2NzgwNjY2YWNlNThlYmNiJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.guDKg7UsRM24inf8r4mH4EBSyFsUiBFZmxfS3P2GhMs)
> 
> For this one - the inner cells should change to cell percents and then the total row would have a row percent and the total column would be a column percent.
> 
> > To clarify the request here, do you want the Signed-in column to be 100%, and then the remaining percentages be the amount of the previous column? For example, the first two rows would be: Baylor Scott and White 14 (100%), 13(92.86), 13(92.86), 13(92.86), 13(92.86), 287 HealthPartners 10,230 (100%), 8,489 (82.98%), 8,252 (97.21%), 8,240 (99.85%), 7,931 (96.25%), 109,912 ![image](https://private-user-images.githubusercontent.com/101136647/367320344-18acf409-f2c6-4a8b-941c-7c75063f33ee.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjYyNDk0MDYsIm5iZiI6MTcyNjI0OTEwNiwicGF0aCI6Ii8xMDExMzY2NDcvMzY3MzIwMzQ0LTE4YWNmNDA5LWYyYzYtNGE4Yi05NDFjLTdjNzUwNjNmMzNlZS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwOTEzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDkxM1QxNzM4MjZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0wMzFjOTVkMWVmMzI5YTk2OTYyNTdlMDlkMWIzNmQwNTJmYTNmZDFlZWRjNjFmZGQwOWJmZjE3MmYyNmM2ZTMxJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.PfU0A069HzdqYMveDLZ3bNKyzzv-CpGg16kdOaB5tXI)
> 
> For this one, yes, that sounds exactly right!

@robertsamm  How does this look? I thought because Signed-In will be 100%, and the rest a percentage of Signed-In, that Total Recruits shouldn't have a percentage. Please let me know if that needs to be changed

![image](https://github.com/user-attachments/assets/b1d2b14b-1b62-4ad4-9401-c9e4ca1a110c)

**Comment by mnataraj92** [comment]
> @mnataraj92 Do you know why these numbers may be off or if we're missing something in the logic? It looks like it's mostly a HF problem. Not sure if the QC reflects this as well
> 
> * **Automated Verification**: verified=yes, RcrtV_Automated_v1r0=method used, RcrtV_Mannual_v1r0= method NOT used, RcrtV_Outreach_v1r0 = No/NULL
> * **Manual Verification**: verified=yes, RcrtV_Mannual_v1r0= method used, RcrtV_Outreach_v1r0 =NO/Null
> * **Manual Verification and Outreach**: verified=yes, RcrtV_Mannual_v1r0= method used, RcrtV_Outreach_v1r0=yes
> 
> ![image](https://private-user-images.githubusercontent.com/101136647/371240731-88782442-a4f8-45da-8492-363e8c0a3426.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjc5NjM4MjMsIm5iZiI6MTcyNzk2MzUyMywicGF0aCI6Ii8xMDExMzY2NDcvMzcxMjQwNzMxLTg4NzgyNDQyLWE0ZjgtNDVkYS04NDkyLTM2M2U4YzBhMzQyNi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMDAzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTAwM1QxMzUyMDNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0zZDI1MTI0M2MxNjg0Y2Q3YzlkZDU2ZTUyOGEzMTc3MjM1NTExMGJiNmNiOWVhNzg5MDZlOWQyMGZmZGI1MjIwJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.M0SEAOOzv5J8m5Uajg37inWxQb0ixrjSFgXacEwPcXQ)

Hi @KELSEYDOWLING7 it looks like that those that fall under automated verification might be showing up as missing in this screenshot for some reason? I just looked at the data in BQ and it looks like as of this morning there are 6408 verified individuals from HFH that used automated verification. There are a handful of individuals that are verified but missing verification mode (n = 38) but that's because they had data destroyed so that is expected. There is one individual that is verified (looks like they were verified this morning) from HFH that does not yet have a verification mode, but that might be because BQ has not yet refreshed- I'll keep an eye on this one to make sure that verification mode is sent.

**Comment by KELSEYDOWLING7** [comment]
@mnataraj92 Ok got it. How are you defining automated verification? I'm thinking there's an issue with the conditionality we have for just HF

**Comment by mnataraj92** [comment]
d_821247024 (verification) = 197316935 (verified) & state_d_444699761 (automated verification) = 426360242 (method used)

**Comment by KELSEYDOWLING7** [comment]
Hmm ok that's what we have, except we also add RcrtV_Mannual_v1r0= method NOT used and RcrtV_Outreach_v1r0 = No/NULL. I would imagine that should be the case, but I can try removing that extra conditionality and seeing if it changes

**Comment by mnataraj92** [comment]
i think that's it! they can still use manual verification in addition to automated verification

**Comment by KELSEYDOWLING7** [comment]
Ah! Ok thank you! 

@robertsamm is it ok if the people that choose both automated and manual are just counted in the automated column?

**Comment by robertsamm** [comment]
Hi @KELSEYDOWLING7 and @mnataraj92 for this table, we want to have each participant only counted once with the "overall" or final method used. While a pt could have both automated and manual verification flags set to yes, we only want them to appear in the automated column if manual verif was not used. Sites typically run most pts thru the automated pathway first and if they fail the run then run them through the manual. I think if people that choose both automated and manual are put in the automated column they would be double counted bc their true outcome in manual verification.

I believe the coding should be something like this but please review for accuracy
automated: automated=yes AND manual=no AND outreach required = no/null
manual: automated yes or no AND manual = yes AND outreach required = no/null
manual verification with outreach: automated = yes or no AND manual = yes AND outreach required = yes

**Comment by KELSEYDOWLING7** [comment]
So the way my code is set up it's not possible for a participant to be double counted. It's essentially an ifelse structure. 

This is what it looks like with the logic from Madhuri's suggestion: 
If automated=yes, then Automated Verification
ELSE IF manual=yes and outreach=no or NULL, then Manual Verification
ELSE IF manual=yes & outreach=yes, then Manual Verification Requiring Outreach
![image](https://github.com/user-attachments/assets/23d8c5f8-65a7-42a6-b838-3c969da34810)


Compared to the current table:
![image](https://github.com/user-attachments/assets/7ac1ef69-3a79-4d79-bedf-c34b0fb9ba93)


I can try that logic you just mentioned, but that logic for automatic is the same as what we have currently, which is counting no one in HF. Also with that logic those that have both automated and manual would be counted under manual instead of automated

**Comment by robertsamm** [comment]
@KELSEYDOWLING7 HFH, HP, and UC don't have any participants who have completed manual verification in the first logic which seems low. What if you flipped it the order?
So did:
If manual=yes & outreach=yes, then Manual Verification Requiring Outreach
ELSE IF manual=yes and outreach=no or NULL, then Manual Verification
ELSE IF automated=yes, then Automated Verification

**Comment by KELSEYDOWLING7** [comment]
That could work! That looks like this:

![image](https://github.com/user-attachments/assets/4a7205a8-6d9e-49c1-b4f2-a09501503db0)

**Comment by robertsamm** [comment]
@KELSEYDOWLING7 yes, I think that prob looks right! Thanks for working through this one!!

So interesting to see how wide the range is of sites requiring manual verification. I think it does make sense though as some sites require all passive recruits go through manual. Good that the counts/percents of outreach remains low though.

**Comment by KELSEYDOWLING7** [comment]
Great, well that's all the issues then! This will all be applied to Monday's report. Am I good to close this issue?

**Comment by robertsamm** [comment]
Yes, closed!

**Comment by robertsamm** [comment]
Reopening this issue to follow up on the verification mode metrics. I reviewed this with Michelle and Kaitlyn in our weekly reports review mtg, and we wanted to do a little more digging before we send this to the sites to confirm the numbers look accurate. 

All 3 verification methods (automatic, manual, outreach required) are showing in the DD as required so there shouldn't be many null values. We've also received reports in the past from the sites that the process for outreach required takes a lot of work on the staff end, so we are surprised by the low counts for some of those sites in the 'outreach required' category. We assumed those numbers would be higher since the sites have mentioned how burdensome it is.

@KELSEYDOWLING7 and @mnataraj92 we have a requests below to look into the data more in BQ. These can just be basic frequencies pulled from BQ, no need to put them in the actual Ops Report.

1. Could you please provide a basic frequency count of each verification method category for when the flags = yes, no, or null and stratify it by site? 

With this we'll be able to see if the sites have a lot of null values when they shouldn't be, which could be affecting the counts and also be able to confirm that the 'outreach required' counts are accurate.

**Comment by KELSEYDOWLING7** [comment]
[Verification Method Counts.csv](https://github.com/user-attachments/files/17263193/Verification.Method.Counts.csv)

**Comment by robertsamm** [comment]
Thanks @KELSEYDOWLING7 for running these frequencies! 

From the tables linked above, it looks like there aren't too many participants who have null values for 'automatic method' and 'manual method', but have ~6,000 participants with 'outreach required' as 'null'. @mnataraj92 when I checked the DD it looked like 'outreach required method' is a required variable, so it seems like we shouldn't have that many null values. But can you confirm that it's required?

@brotzmanmj it looks like the counts in the metric for outreach required=yes are accurate based on the counts of the flags in the table linked above. The numbers are still pretty low across most sites. With less than 400 participants in the cohort requiring outreach at all. Once we finalize this metric and whether outreach should be null for so many people, we could ask the sites to review and let us know if this seems accurate from their end?

@KELSEYDOWLING7 sorry for additional changes! Could we update the category labels in the table on the Ops Report to the following below to make it more clear we aren't double counting people who may have multiple methods?
'Automated Verification Only'
'Manual Verification w/out Outreach'
'Manual Verification w/ Outreach'

**Comment by mnataraj92** [comment]
Hi @robertsamm, state_d_188797763 (RcrtV_Outreach_v1r0) is a required variable according to the dictionary. I'm wondering if sites are maybe only sending it if manual verification is used since this variable can only equal yes when manual verification is used? And that's why there's so many null?

**Comment by KELSEYDOWLING7** [comment]
That's ok! The renaming is a quick fix. Do you need today's report redacted or is it ok to just update next week's report?

**Comment by robertsamm** [comment]
@KELSEYDOWLING7 Next week's report is fine! Thanks Kelsey!

**Comment by robertsamm** [comment]
@mnataraj92 that makes sense, but do we need to change this and have the sites send 'no'/'not used' even when automatic verification is the only method used? Or leave as is?

**Comment by brotzmanmj** [comment]
I believe Nicole's intention in making all three variables required was that we would have a yes/no response for all of them to indicate which were used/attempted.

Looking at the separate frequencies of the three variables by site, it seems there is substantial inconsistency in how they are interpreting and coding these variables. I don't know if we want them to change course but we should discuss with Nicole if we want them all to conform to a standard or let them continue to self-define. Madhuri can you set up a quick meeting for us with Nicole when you can? 30 minutes should be enough

**Comment by mnataraj92** [comment]
@brotzmanmj sure!

**Comment by robertsamm** [comment]
Discussed with Nicole, Michelle, and Madhuri and will work with sites to correct any data that may need to be corrected for the verification methods they are sending. However, the metric itself is coded correctly now and does not need to be changed so closing this issue. Separate issues will be creating on the site github repo for fixing site data.
